airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py

```diff
@@ -101,4 +101,8 @@ def get_request_body_json(
 
     def stream_slices(self) -> Iterable[StreamSlice]:
         sub_slices = (s.stream_slices() for s in self.stream_slicers)
-        return (dict(ChainMap(*a)) for a in itertools.product(*sub_slices))
+        product = itertools.product(*sub_slices)
+        for stream_slice_tuple in product:
+            partition = dict(ChainMap(*[s.partition for s in stream_slice_tuple]))
+            cursor_slice = dict(ChainMap(*[s.cursor_slice for s in stream_slice_tuple]))
```

Comment:
It feels weird that we would have multiple cursor here but I guess it has always been like that. I'm just wondering if we should fail loudly because we don't understand this business case or just let it happen and have an undetermined behavior

Reply:
agreed. fixed